src/app/services/resource.service.ts

```diff
@@ -63,27 +80,55 @@ export class ResourceService {
+    switch (this.endpointMode) {
+      case EndpointMode.TRANSMART: {
+        return this.transmartResourceService.getTreeNodes(root, depth, hasCounts, hasTags);
+      }
+      default: {
+        return this.transmartResourceService.getTreeNodes(root, depth, hasCounts, hasTags);
+      }
+    }
   }
 
   // -------------------------------------- count calls --------------------------------------
   updateInclusionExclusionCounts(constraint: Constraint,
                                  inclusionConstraint: Constraint,
                                  exclusionConstraint?: Constraint): Promise<any> {
     return new Promise<any>((resolve, reject) => {
-      this.transmartResourceService.updateInclusionExclusionCounts(constraint, inclusionConstraint, exclusionConstraint)
-        .then(() => {
-          this.inclusionCounts =
-            TransmartMapper.mapTransmartCountItem(this.transmartResourceService.inclusionCounts);
-          this.exclusionCounts =
-            TransmartMapper.mapTransmartCountItem(this.transmartResourceService.exclusionCounts);
-          this.selectedStudyConceptCountMap =
-            TransmartMapper.mapStudyConceptCountObject(this.transmartResourceService.studyConceptCountObject);
-          resolve(true);
-        })
-        .catch(err => {
-          reject(err);
-        });
+      switch (this.endpointMode) {
+        case EndpointMode.TRANSMART: {
+          this.transmartResourceService.updateInclusionExclusionCounts(constraint, inclusionConstraint, exclusionConstraint)
+            .then(() => {
+              this.inclusionCounts =
+                TransmartMapper.mapTransmartCountItem(this.transmartResourceService.inclusionCounts);
+              this.exclusionCounts =
+                TransmartMapper.mapTransmartCountItem(this.transmartResourceService.exclusionCounts);
+              this.selectedStudyConceptCountMap =
+                TransmartMapper.mapStudyConceptCountObject(this.transmartResourceService.studyConceptCountObject);
+              resolve(true);
+            })
+            .catch(err => {
+              reject(err);
+            });
+          break;
+        }
+        default: {
+          this.transmartResourceService.updateInclusionExclusionCounts(constraint, inclusionConstraint, exclusionConstraint)
+            .then(() => {
+              this.inclusionCounts =
+                TransmartMapper.mapTransmartCountItem(this.transmartResourceService.inclusionCounts);
+              this.exclusionCounts =
+                TransmartMapper.mapTransmartCountItem(this.transmartResourceService.exclusionCounts);
+              this.selectedStudyConceptCountMap =
+                TransmartMapper.mapStudyConceptCountObject(this.transmartResourceService.studyConceptCountObject);
+              resolve(true);
+            })
+            .catch(err => {
+              reject(err);
+            });
+          break;
+        }
+      }
```

Comment:
this `switch case Transmart:  transmartResource...; default:  transmartResource...` way requires a lot of code duplications. Maybe this could be replaced by something like endpointResource interface?

Reply:
what do you mean?